### PR TITLE
Setup codebase for Notifications within the plugin.

### PR DESCRIPTION
**Pull request by retrevor**
- Enables notifications when files are uploaded (aka decentralized) to Swarm. 
- The message is set in the code in a new class: /lib/Notifications/Notifier.php. 
- The notification is sent to the user who uploads the content. 

- Added the Appname 'files_external_ethswarm' as a constant in a new class for application-wide use.

### CODE REVIEW

File: appinfo/info.xml [hunk 1]
```diff
@@ -63,7 +63,7 @@ Experience the future of data storage with 5GB of free, decentralized storage on
 
 [Get your free licence here!](https://metaprovide.org/hejbit/start) .
     ]]></description>
-    <version>0.5.4</version>
+    <version>0.5.0</version>
```
Comment (mahiarirani) [inline appinfo/info.xml:66]: Is there any reason to update the version like this for this PR?

File: lib/Backend/BeeSwarm.php [hunk 2]
```diff
@@ -69,10 +68,10 @@ public function __construct(string $appName, IL10N $l, IConfig $config, LoggerIn
 		$this->logger = $logger;
 		$this->globalStoragesService = $globalStoragesService;
 		$this
-			->setIdentifier('files_external_ethswarm')
+			->setIdentifier(AppConstants::APP_NAME)
 			->addIdentifierAlias('\OC\Files\External_Storage\BeeSwarm') // legacy compat
 			->setStorageClass('\OCA\Files_External_Ethswarm\Storage\BeeSwarm')
-			->setText($l->t('Swarm By Hejbit'))
+			->setText($l->t('Swarm'))
```
Comment (mahiarirani) [inline lib/Backend/BeeSwarm.php:74]: I think @JoaoSRaposo wanted to have the HejBit in name in the settings page.

File: lib/Storage/BeeSwarm.php [hunk 3]
```diff
@@ -425,16 +445,6 @@ public function getMetaData($path)
 		return $data;
 	}
 
-	public function getDirectoryContent($path): \Traversable
```
Comment (mahiarirani) [inline lib/Storage/BeeSwarm.php:428]: These changes were for #65 and #68. Is there any reason to revert them?

File: lib/Storage/BeeSwarmTrait.php [hunk 4]
```diff
@@ -104,7 +104,7 @@ private function uploadStream(string $path, string $tmpfile, string $mimetype, i
 	 */
 	private function downloadStream(string $reference)
 	{
-		$endpoint = $this->api_url . DIRECTORY_SEPARATOR . 'bzz' . DIRECTORY_SEPARATOR . $reference . DIRECTORY_SEPARATOR;
```
Comment (mahiarirani) [inline lib/Storage/BeeSwarmTrait.php:107]: This is also related to #63 and not be reverted.

**Review by mahiarirani — COMMENTED** [review]
I believe your changes were not in sync with the latest code, so some changes are reverting other PRs changes. Otherwise, the notification changes look good code wise to me.

File: appinfo/info.xml (hunk 1 above)
Comment (JoaoSRaposo) [reply appinfo/info.xml:66]: The version should be higher than the lastest that we publish. I think @frozenbanana controls the app versionning on the release process.

File: lib/Backend/BeeSwarm.php (hunk 2 above)
Comment (JoaoSRaposo) [reply lib/Backend/BeeSwarm.php:74]: Yes :)

File: lib/Storage/BeeSwarm.php (hunk 3 above)
Comment (JoaoSRaposo) [reply lib/Storage/BeeSwarm.php:428]: I think we should mantain @Thesola10 version on this one

Comment (retrevor) [reply lib/Storage/BeeSwarm.php:428]: Agreed, the original state remain.

File: lib/Storage/BeeSwarm.php [hunk 5]
```diff
@@ -427,9 +444,7 @@ public function getMetaData($path)
 
 	public function getDirectoryContent($path): \Traversable
 	{
-		$rows = $this->filemapper->getPathTree($path, $this->storageId,
-		                                       incSelf: false,
-		                                       recursive: false);
+		$rows = $this->filemapper->getPathTree($path, $this->storageId, false);
```
Comment (mahiarirani) [inline lib/Storage/BeeSwarm.php:447]: @retrevor would you please verify this line change is related to your work? I think it's also still related to #68 and shouldn't be reverted

**Review by mahiarirani — CHANGES_REQUESTED** [review]

File: appinfo/info.xml (hunk 1 above)
Comment (retrevor) [reply appinfo/info.xml]: Agreed, this should only be changed on the final release/deploy. Change has been removed

**Pull request merged**